src/playground/service.clj

```diff
@@ -92,6 +95,25 @@
                     interceptor-chain/terminate)
                 )))})
 
+(def author-interceptor
+  "throw unauthorized 403 by author
+  (allows admin and author of entry)"
+  {:name ::author-interceptor
+   :enter (fn [context]
+            (let [role     (get-in context [:request :session :identity :role])
+                  username (get-in context [:request :session :identity :username])
+                  id       (get-in context [:request :path-params :id])
+                  db       (get-in context [:request :db])
+                  author (:author (invoices.retrieve/get-author (:request context)))]
```

Comment:
Different endpoints may have different ways of deriving its corresponding author.

But here you are using `invoices.retrieve` to derive the author for every single consumer of `author-interceptor`.

i.e. we tie a specific logic to the general behavior, which is wrong.

instead, _parameterize_ `author-interceptor`, i.e. make it a `defn` which accepts a `fn` as its argument:

![image](https://user-images.githubusercontent.com/1162994/57176088-d49e5d00-6e53-11e9-9028-7eb7b68cf22a.png)